src/datepicker/datepicker.component.ts

```diff
@@ -177,16 +177,19 @@ export class DatePicker implements OnDestroy, AfterViewInit, OnChanges {
 			if (changes.value && this.didDateValueChange(changes.value.currentValue, changes.value.previousValue)) {
 				dates = changes.value.currentValue;
 			}
+			// only reset the flatpickr instance on Input changes
 			this.flatpickrInstance = flatpickr(`#${this.id}`, this.flatpickrOptions);
-			this.flatpickrInstance.setDate(dates);
 			this.setDateValues(dates);
 		}
 	}
 
 	ngAfterViewInit() {
 		this.flatpickrInstance = flatpickr(`#${this.id}`, this.flatpickrOptions);
+		if (this.value === null) {
+			this.value = [];
```

Comment:
Isn't it already set to this when it is initialized?

Reply:
It is initialized to `[]`, however `ngModel` and friends can override that with an explicit `null`, so in this case we just reset it (since nothing else in the component is set up to handle `null`s)

**However** you bring up a good point ... I'll change this (and the other comment) out to use a setter/getter on the input instead so we can easilly catch any instance of `null` value passing